Providers/Profile.swift

```diff
@@ -296,7 +296,7 @@ public class BrowserProfile: Profile {
         let commands = items.map { item in
             SyncCommand.fromShareItem(item, withAction: "displayURI")
         }
-        self.remoteClientsAndTabs.insertCommands(commands, forClients: clients)
+        self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> { self.syncManager.syncClients() }
```

Comment:
Yes. 

```
    private func getSyncDelegate() -> SyncDelegate {
        if let app = self.app {
            return BrowserProfileSyncDelegate(app: app)
        }
        return CommandDiscardingSyncDelegate()
    }
```

There are various approaches to fixing this.

* Replace `CommandDiscardingSyncDelegate` with one that stuffs incoming tabs into the “View Later” queue (which didn't exist when it was written). This works around the need to have access to the `UIApplication` in order to process incoming commands, and partly eliminates the distinction between in-extension and in-app syncs.
* Don't advance client timestamps when syncing clients from an extension. This is kinda hacky. We'll re-process our own record.
* Expose a way to do an upload-only clients sync. "Upload-only" is a bit of a misnomer, granted.
* Don't trigger an outgoing sync if we're not sending from inside Firefox. That defeats some of the point of this PR.